File: Python/Program/Motorcycle.py
from Vehicle import Vehicle

class Motorcycle(Vehicle):
    __Tipe = ""
    __KapasitasTangki = 0
    
    def __init__(self, NoPlat, Merk, TahunProduksi, Warna, Tipe, KapasitasTangki):
        super().__init__(NoPlat, Merk, TahunProduksi, Warna)
        self.__Tipe = Tipe
        self.__KapasitasTangki = KapasitasTangki
        
    # setter getter
    
    def setTipe(self, Tipe):
        self.__Tipe = Tipe
        
    def getTipe(self):
        return self.__Tipe
    
    def setKapasitasTangki(self, KapasitasTangki):
        self.__KapasitasTangki = KapasitasTangki
        
    def getKapasitasTangki(self):
        return self.__KapasitasTangki

File: CPP/Program/Motorcycle.cpp
#pragma once

#include <string>
#include <iostream>
#include "Vehicle.cpp"

using namespace std;

class Motorcycle : public Vehicle {
private:
    string Tipe;
    int KapasitasTangki;

public:
    Motorcycle(string NoPlat, string Merk, int TahunProduksi, string Warna,
               string Tipe, int KapasitasTangki)
        : Vehicle(NoPlat, Merk, TahunProduksi, Warna), Tipe(Tipe), KapasitasTangki(KapasitasTangki) {}

    void setTipe(string Tipe) {
        this->Tipe = Tipe;
    }

    string getTipe() {
        return Tipe;
    }

    void setKapasitasTangki(int KapasitasTangki) {
        this->KapasitasTangki = KapasitasTangki;
    }

    int getKapasitasTangki() {
        return KapasitasTangki;
    }

    void printDetails() const override {
        cout << "Motorcycle details:\n";
        cout << "Plat Nomor : " << getNoPlat() << "\n";
        cout << "Merk : " << getMerk() << "\n";
        cout << "Tahun Produksi : " << getTahunProduksi() << "\n";
        cout << "Warna : " << getWarna() << "\n";
        cout << "Tipe : " << Tipe << "\n";
        cout << "Kapasitas Tangki : " << KapasitasTangki << "\n";
        
    }
};